qa/tasks/cephadm.py

```diff
@@ -116,54 +116,29 @@ def normalize_hostnames(ctx):
 
 
 @contextlib.contextmanager
-def download_cephadm(ctx, config, ref):
+def download_cephadm(ctx, config):
     cluster_name = config['cluster']
 
+    os_type = teuthology.get_distro(ctx)
+    os_version = teuthology.get_distro_version(ctx)
+
+    branch = config.get('branch', 'master')
+    sha1 = config.get('sha1', 'latest')
+    flavor = config.get('flavor', 'default')
+
     if config.get('cephadm_mode') != 'cephadm-package':
-        ref = config.get('cephadm_branch', ref)
-        git_url = config.get('cephadm_git_url', teuth_config.get_ceph_git_url())
-        log.info('Downloading cephadm (repo %s ref %s)...' % (git_url, ref))
-        if ctx.config.get('redhat'):
-            log.info("Install cephadm using RPM")
-            # cephadm already installed from redhat.install task
-            ctx.cluster.run(
-                args=[
-                    'cp',
-                    run.Raw('$(which cephadm)'),
-                    ctx.cephadm,
-                    run.Raw('&&'),
-                    'ls', '-l',
-                    ctx.cephadm,
-                ]
-            )
-        elif git_url.startswith('https://github.com/'):
-            # git archive doesn't like https:// URLs, which we use with github.
-            rest = git_url.split('https://github.com/', 1)[1]
-            rest = re.sub(r'\.git/?$', '', rest).strip() # no .git suffix
-            ctx.cluster.run(
-                args=[
-                    'curl', '--silent',
-                    'https://raw.githubusercontent.com/' + rest + '/' + ref + '/src/cephadm/cephadm',
-                    run.Raw('>'),
-                    ctx.cephadm,
-                    run.Raw('&&'),
-                    'ls', '-l',
-                    ctx.cephadm,
-                ],
-            )
-        else:
-            ctx.cluster.run(
-                args=[
-                    'git', 'archive',
-                    '--remote=' + git_url,
-                    ref,
-                    'src/cephadm/cephadm',
-                    run.Raw('|'),
-                    'tar', '-xO', 'src/cephadm/cephadm',
-                    run.Raw('>'),
-                    ctx.cephadm,
-                ],
-            )
+        chacra.pull(
+            project='ceph',
+            distro=os_type,
+            release=os_version,
+            arch='x86_64',  # TODO: config contains the arch?
```

Comment:
Try to get the target arch from the context and any remote node where you are going to run the image.

Reply:
Also, keep in mind that `arch` record can be provided within the config and if it is ignore, someone who gonna refactor this code in future may want to know why.